### PR TITLE
feat: add getstream

### LINKED ISSUE #4679: 🚀 Feature: Add GetStream as a Chat Provider
### 🔖 Feature description

Use the [GetStream docs](https://getstream.io/chat/docs/node/?language=javascript)  to create a new chat provider while following [Novu's guide to a new provider](https://docs.novu.co/community/add-a-new-provider#how-to-add-a-new-provider)

### 🎤 Why is this feature needed ?

more options for the community

### ✌️ How do you aim to achieve this?

community contribution

### 🔄️ Additional Information

_No response_

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

### PR DESCRIPTION
**Pull request by vikaspatil0021**
### What change does this PR introduce?
fixes #4679

Added GetStream as chat provider

### CODE REVIEW

File: providers/getstream/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Novu Getstream Provider
+
+A Getstream chat provider library for [@novu/node](https://github.com/novuhq/novu)
+
+## Usage
+
+```javascript
+import { GetstreamChatProvider } from '@novu/getstream';
+
+const provider = new GetstreamChatProvider({});
```
Comment (Cliftonz) [inline providers/getstream/README.md:10]: this should contain the API key as defined in the provider

File: providers/getstream/package.json [hunk 2]
```diff
@@ -0,0 +1,78 @@
+{
+  "name": "@novu/getstream",
+  "version": "0.16.3",
```
Comment (Cliftonz) [inline providers/getstream/package.json:3]: this should be 0.21.0

File: packages/application-generic/package.json [hunk 3]
```diff
@@ -63,6 +63,7 @@
     "@novu/forty-six-elks": "^0.21.0",
     "@novu/generic-sms": "^0.21.0",
     "@novu/gupshup": "^0.21.0",
+    "@novu/getstream": "^0.16.3",
```
Comment (Cliftonz) [inline packages/application-generic/package.json:66]: This should be 0.21.0

**Review by Cliftonz — CHANGES_REQUESTED** [review]

**Comment by vikaspatil0021** [comment]
Hey @Cliftonz  Requested changes have been updated :)

**Comment by Cliftonz** [comment]
@vikaspatil0021 Looks like we need to resolve some conflicts, please be patient we are merging alot of providers and may need to do this a few time.

**Comment by Cliftonz** [comment]
Can you resolve these conflicts?

**Comment by vikaspatil0021** [comment]
@Cliftonz Yes, I can resolve them

**Comment by Cliftonz** [comment]
@vikaspatil0021 Looks like we need to resolve some conflicts again, please be patient we are merging alot of providers and may need to do this a few time.

**Comment by Cliftonz** [comment]
@vikaspatil0021 Looks like we need to resolve some conflicts again, please be patient we are merging alot of providers and may need to do this a few time.

**Comment by vikaspatil0021** [comment]
Hey @Cliftonz , any updates?

**Comment by Cliftonz** [comment]
@vikaspatil0021 Hey things have stabilized, if you fix the conflicts listed here and @ me again I should be able to get it merged.

**Comment by vikaspatil0021** [comment]
Hi @Cliftonz, I've fixed the conflicts. Can you please check it?

**Comment by Cliftonz** [comment]
@vikaspatil0021 looks like we have a few unknow words in this pr. Can you fix them?
https://github.com/novuhq/novu/actions/runs/6976284336/job/18996272086?pr=4692

**Comment by vikaspatil0021** [comment]
Hi @Cliftonz, I've updated the `.cspell.json` file by including both "getstream" and "Getstream" in the list of recognized words. Kindly review and confirm.

**Comment by vikaspatil0021** [comment]
Hey @Cliftonz , could you please review and approve the workflows again? I believe there is one final check that needs to be fixed.

**Comment by vikaspatil0021** [comment]
@Cliftonz  I've made modifications to the files **digest-events.e2e.ts** and **commands.ts**. Kindly reconsider approving the workflow once again.

**Comment by vikaspatil0021** [comment]
@Cliftonz  I've made more changes to the **digest-events.e2e.ts** and **user-preference.spec.ts** files. Please review and approve these modifications once again.

**Pull request merged**